File: hexameter-core/src/main/java/org/codetome/hexameter/core/api/contract/SatelliteData.java
package org.codetome.hexameter.core.api.contract;

import org.codetome.hexameter.core.api.Hexagon;

/**
 * Represents arbitrary data which can be attached to a Hexagon.
 * An implementation should contain a set of fields for advanced
 * grid algorithms like pathfinding.
 */
public interface SatelliteData {

    /**
     * Tells whether the Hexagon can be passed over when moving around the map or not.
     *
     * @return is passable?
     */
    boolean isPassable();

    /**
     * Sets whether the Hexagon is passable or not.
     *
     * @param passable passable?
     */
    void setPassable(boolean passable);

    /**
     * @return true if the {@link Hexagon} can bee seen through, false otherwise.
     */
    boolean isOpaque();

    /**
     * Set whether the {@link Hexagon} can be seen through.
     *
     * @param opaque is opaque?
     */
    void setOpaque(boolean opaque);

    /**
     * Returns the movement cost when moving over the Hexagon.
     *
     * @return movement cost
     */
    double getMovementCost();

    /**
     * Sets the movement cost.
     *
     * @param movementCost movement cost
     */
    void setMovementCost(double movementCost);

}

File: mixite2.core/src/commonMain/kotlin/org/hexworks/mixite2/core/api/contract/SatelliteData.kt
package org.hexworks.mixite2.core.api.contract

/**
 * Represents arbitrary data which can be attached to a Hexagon.
 * An implementation should contain a set of fields for advanced
 * grid algorithms like pathfinding.
 */
interface SatelliteData {

    /**
     * Tells whether the Hexagon can be passed over when moving around the map or not.
     *
     * @return is passable?
     */
    /**
     * Sets whether the Hexagon is passable or not.
     *
     * @param passable passable?
     */
    var isPassable: Boolean

    /**
     * @return true if the [Hexagon] can bee seen through, false otherwise.
     */
    /**
     * Set whether the [Hexagon] can be seen through.
     *
     * @param opaque is opaque?
     */
    var isOpaque: Boolean

    /**
     * Returns the movement cost when moving over the Hexagon.
     *
     * @return movement cost
     */
    /**
     * Sets the movement cost.
     *
     * @param movementCost movement cost
     */
    var movementCost: Double

}
